File: Python/6kyu/SumOfDigits.py
from math import floor


import math


def digital_root(n):

    if n > 9:
        n = sum([int(l) for l in list(str(n))])
        n = digital_root(n)

    return n


def digital_root_kata(n):
    return n if n < 10 else digital_root(sum(map(int, n)))


print(digital_root(942))

File: CSharp/6kyu/SumOfDigits.cs
using System.Linq;
using System;

namespace CSharp._6Kyu
{
    public static partial class Kata6
    {
        public static int DigitalRoot(long n)
        {
            int number = n.ToString().ToArray().Sum(x => int.Parse(x.ToString()));

            if (number > 9)
            {
                number = DigitalRoot(number);
            }

            return number;
        }

        public static int DigitalRoot_Kata(long n)
        {
            return (int)(1 + (n - 1) % 9);
        }
    }
}
